coriolis/utils.py

```diff
@@ -77,6 +87,46 @@
 """
 
 
+def _get_local_ips():
+    ifaces = netifaces.interfaces()
+    ret = []
+    for iface in ifaces:
+        if iface == "lo":
+            continue
+        addrs = netifaces.ifaddresses(iface)
+        ret.append(
+            {
+                iface: {
+                    "ipv4": addrs.get(netifaces.AF_INET),
+                    "ipv6": addrs.get(netifaces.AF_INET6),
+                },
+            }
+        )
+    return ret
+
+
+def _get_host_os_info():
+    info = None
+    # This exists on all modern Linux systems
+    if os.path.isfile("/etc/os-release"):
+        with open("/etc/os-release") as fd:
+            info = fd.read().split('\n')
+    return info
+
+
+def get_diagnostics_info():
+    # TODO(gsamfira): decide if we want any other kind of
+    # diagnostics.
+    packages = list(freeze.freeze())
+    return {
+        "application": os.path.basename(main.__file__),
```

Comment:
NIT: if possible, I would attempt to formalize this structure by declaring a dedicated JSON schema for it. (this will greatly aid in providing some "promise of stability" for the UI/CLI efforts for this.

Reply:
Honestly, until we decide if we want to add more stuff here, and what that stuff is, I would just keep it loose. The info returned by this is not (at this point), meant to be parsed automatically by anything. It's something we will use to at a glance know what is running in a particular Coriolis environment (packages, versions, OS, IPs, etc). It's strictly meant for debugging purposes.